### PR TITLE
BUG: DataFrame constructor raised ValueError with list-like data and …

### LINKED ISSUE #30280: Can't initialise DataFrame with ExtensionDtype, but can do so for Series
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import pytz

temp = pd.Series([pd.Timestamp("2019-07-19 07:04:57+0100", tz=pytz.FixedOffset(60))])
pd.Series(["2019-06-19 07:04:57"], dtype=temp.dtype)  # works
pd.DataFrame(["2019-06-19 07:04:57"]).astype(temp.dtype)  # works
pd.DataFrame(["2019-06-19 07:04:57"], dtype=temp.dtype)  # fails
```
#### Problem description

I get the following error:
```
ValueError: failed to cast to 'datetime64[ns, pytz.FixedOffset(60)]' (Exception was: data type not understood)
```

#### Expected Output

I would have expected the same output as I got in the first two examples above labelled `#works`.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-70-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 0.25.3
numpy            : 1.17.4
pytz             : 2019.3
dateutil         : 2.8.0
pip              : 19.3.1
setuptools       : 41.4.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 7.9.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by simonjayhawkins**
…dtype specified

- [ ] closes #30280
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
That was fast, thanks!

File: pandas/tests/extension/base/constructors.py [hunk 1]
```diff
@@ -64,6 +64,15 @@ def test_from_dtype(self, data):
         result = pd.Series(list(data), dtype=str(dtype))
         self.assert_series_equal(result, expected)
 
+        # gh-30280
+
+        expected = pd.DataFrame(data).astype(dtype)
```
Comment (WillAyd) [inline pandas/tests/extension/base/constructors.py]: Can this test be parametrized instead? Or is the `.astype` call a hanging point?

File: pandas/core/internals/construction.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def init_ndarray(values, index, columns, dtype=None, copy=False):
 
         index, columns = _get_axes(len(values), 1, index, columns)
         return arrays_to_mgr([values], columns, index, columns, dtype=dtype)
-    elif is_extension_array_dtype(values):
+    elif is_extension_array_dtype(values) or is_extension_array_dtype(dtype):
```
Comment (WillAyd) [inline pandas/core/internals/construction.py]: I'm not super well versed in this area but do you know what was actually throwing the error? Just wondering if there's a more generic fix to be made

**Review by WillAyd — COMMENTED** [review]
Thanks for the PR

File: pandas/tests/extension/base/constructors.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/extension/base/constructors.py]: > Can this test be parametrized instead?

maybe could parametrize in a follow-on if this test is OK.

> Or is the `.astype` call a hanging point?

and also the assert method changes, so the four tests could end up with less clarity. maybe could split the test, but that would result in additional skips required in _numpy and json tests.

File: pandas/core/internals/construction.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/internals/construction.py]: This fix is consistent with the handling of Categorical in the preceding condition.

**Review by WillAyd — APPROVED** [review]
Thanks for clarifications - all valid points so lgtm

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**